ouroboros-consensus/src/Ouroboros/Storage/LedgerDB/InMemory.hs

```diff
@@ -268,6 +271,55 @@ ledgerDbTail (Snap _ _ ss) = ledgerDbTail ss
 ledgerDbTail (Skip _   ss) = ledgerDbTail ss
 ledgerDbTail (Tail _ cs)   = cs
 
+{-------------------------------------------------------------------------------
+  Result types
+-------------------------------------------------------------------------------}
+
+-- | The result of 'ledgerDbPushMany', i.e. when validating a number of new
+-- blocks.
+--
+-- About the boolean type index: if all blocks given to 'ledgerDbPushMany' are
+-- 'Reapply', and so @ap ~ 'True@, then the result /must/ be 'ValidBlocks'.
+data PushManyResult e l r :: Bool -> * where
```

Comment:
Personally I find it easier to read as
```suggestion
data PushManyResult e l r (k :: Bool) where
```
But happy for this to be ignored!

Reply:
I see. My thinking was: name type *parameters* but not type *indices*. Wasn't this a thing in Coq or Agda? I don't remember...